misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/action/RemoteMethodCallActionEvaluator.java

```diff
@@ -0,0 +1,154 @@
+import java.util.Optional;
+
+import static org.ballerinalang.debugadapter.evaluation.engine.InvocationArgProcessor.generateNamedArgs;
+
+/**
+ * Evaluator implementation for remote method call invocation actions.
+ *
+ * @since 2.0.0
+ */
+public class RemoteMethodCallActionEvaluator extends MethodCallExpressionEvaluator {
+
+    private final RemoteMethodCallActionNode syntaxNode;
+    private final String methodName;
+    private final Evaluator objectExpressionEvaluator;
+    private final List<Map.Entry<String, Evaluator>> argEvaluators;
+
+    public RemoteMethodCallActionEvaluator(SuspendedContext context, RemoteMethodCallActionNode remoteMethodActionNode,
+                                           Evaluator expression, List<Map.Entry<String, Evaluator>> argEvaluators) {
+        super(context, null, expression, argEvaluators);
+        this.syntaxNode = remoteMethodActionNode;
+        this.objectExpressionEvaluator = expression;
+        this.argEvaluators = argEvaluators;
+        this.methodName = syntaxNode.methodName().toSourceCode().trim();
+    }
+
+    @Override
+    public BExpressionValue evaluate() throws EvaluationException {
+        try {
+            // Calls a remote method of a client object. This works the same as a method call expression, except that
+            // it is used for a client object method with the remote qualifier.
+            BExpressionValue result = objectExpressionEvaluator.evaluate();
+            BVariable resultVar = VariableFactory.getVariable(context, result.getJdiValue());
+
+            // If the expression result is an object, try invoking as an object method invocation.
+            if (result.getType() != BVariableType.OBJECT) {
+                throw new EvaluationException(String.format(EvaluationExceptionKind.CUSTOM_ERROR.getString(),
+                        "invalid remote method call: expected a client object, but found 'other'"));
+            }
+
+            Value invocationResult = invokeRemoteMethod(resultVar);
+            return new BExpressionValue(context, invocationResult);
+        } catch (EvaluationException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new EvaluationException(String.format(EvaluationExceptionKind.INTERNAL_ERROR.getString(),
+                    syntaxNode.toSourceCode().trim()));
+        }
+    }
+
+    private Value invokeRemoteMethod(BVariable resultVar) throws EvaluationException {
+
+        ClassDefinitionResolver classDefResolver = new ClassDefinitionResolver(context);
+        String className = resultVar.getDapVariable().getValue();
+        Optional<ClassSymbol> classDef = classDefResolver.findBalClassDefWithinModule(className);
+        if (classDef.isEmpty()) {
+            // Resolves the JNI signature to see if the the object/class is defined with a dependency module.
+            String signature = resultVar.getJvmValue().type().signature();
+            if (!signature.startsWith("L")) {
```

Comment:
What are the possible values can `signature` get here?
Do the remote method signature always starts with `L`?

Reply:
Please refer JNI signature documentation. http://journals.ecs.soton.ac.uk/java/tutorial/native1.1/implementing/method.html#:~:text=Forming%20the%20Method%20Name%20and,name%20in%20UTF%2D8%20form.

Reply:
![Screenshot from 2021-06-11 08-47-00](https://user-images.githubusercontent.com/29032600/121626072-5c72de80-ca92-11eb-8b07-4435046a523e.png)

Reply:
Ack!